datalad/crawler/dbs/base.py

```diff
@@ -85,7 +87,15 @@ def save(self):
         lgr.debug("Writing %s to %s" % (self.__class__.__name__, self._filepath))
         with open(self._filepath, 'w') as f:
             json.dump(db, f, indent=2, sort_keys=True, separators=(',', ': '))
-        self.repo.git_add(self._filepath)  # stage to be committed
+
+        # stage to be committed:
+        if isinstance(self.repo, AnnexRepo):
+            self.repo.add(self._filepath, git=True)
+        elif isinstance(self.repo, GitRepo):
+            self.repo.add(self._filepath)
+        else:
+            raise ValueError("Unknown repo: %s" % self.repo)
```

Comment:
ok

2016-06-02 18:24 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/crawler/dbs/base.py
> <https://github.com/datalad/datalad/pull/482#discussion_r65572472>:
>
> > @@ -85,7 +87,15 @@ def save(self):
> >          lgr.debug("Writing %s to %s" % (self.__class__.__name__, self._filepath))
> >          with open(self._filepath, 'w') as f:
> >              json.dump(db, f, indent=2, sort_keys=True, separators=(',', ': '))
> > -        self.repo.git_add(self._filepath)  # stage to be committed
> > +
> > +        # stage to be committed:
> > +        if isinstance(self.repo, AnnexRepo):
> > +            self.repo.add(self._filepath, git=True)
> > +        elif isinstance(self.repo, GitRepo):
> > +            self.repo.add(self._filepath)
> > +        else:
> > +            raise ValueError("Unknown repo: %s" % self.repo)
>
> I would postpone this channeling of **kwargs until we clearly would
> benefit from it. So for now explicit git=True at the level of GitRepo
> would imho be the cleanest way
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/datalad/datalad/pull/482/files/db10b3ecbd88f2606d99b1e0e8bfcba5dae92d85#r65572472>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKAw_fPCi0c-p20vZRRrbkOSH5cGn0q8ks5qHwPXgaJpZM4Ing8e>
> .
>